cmd/fluxctl/format.go

```diff
@@ -29,3 +45,142 @@ func makeExample(examples ...string) string {
 	}
 	return strings.TrimSuffix(buf.String(), "\n")
 }
+
+func outputFormatIsValid(format string) bool {
+	for _, f := range validOutputFormats {
+		if f == format {
+			return true
+		}
+	}
+	return false
+}
+
+// outputImagesJson sends the provided ImageStatus info to the io.Writer in JSON formatting, honoring limits in opts
+func outputImagesJson(images []v6.ImageStatus, out io.Writer, opts *imageListOpts) error {
+	if opts.limit < 1 {
```

Comment:
Making `opts.limit == 0` equal to an infinite amount of results would be an better option in my opinion. This also makes it easier to programmatically make use of the CLI flag when you sometimes do want to limit the result.

Reply:
Agreed - also that's what the usage note says anyway, so... my bad 😸 

Fixed.

Reply:
ahh actually one sec need to do a bit more

Reply:
In addition I would make the returned error a bit more humane to not step on anyone's toes, while also informing them about the check that failed.

My suggestion would be something like the following: 

> opts.limit is less than 0 but only supports non-negative integers

Reply:
Done!